### PR TITLE
feat: get brc-20 tokens API

### PR DESCRIPTION
**Pull request by abdulhaseeb4239**
# 🔘 PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background

Add a function to get the BRC-20 tokens data from Xverse API

Issue Link: https://linear.app/xverseapp/issue/ENG-3500/[tkn-1-mob📱]-☑️-control-brc-20-display-via-manage-tokens-screen

# 🔄 Changes

Does this PR introduce a breaking change?

- [ ] Yes, Incompatible API changes
- [X] No, Adds functionality (backwards compatible)
- [ ] No, Bug fixes (backwards compatible)

Changes:

- Added a new function in api/xverse.ts to get the BRC-20 tokens data from Xverse API

# ✅ Review checklist

Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

**Comment by DuskaT021** [comment]
branch is outdated @abdulhaseeb4239 fv

File: api/xverse.ts [hunk 1]
```diff
@@ -92,6 +92,34 @@ export async function getCoinsInfo(
...
+): Promise<CoinsResponse | null> {
+  const url = `${XVERSE_API_BASE_URL(network)}/v1/brc20/tokens`;
+
+  const params = {
+    currency: fiatCurrency,
+    tickers: tickers,
+  };
+
+  return axios
+    .get<CoinsResponse>(url, { params })
```
Comment (teebszet) [inline api/xverse.ts:114]: this won't be CoinsResponse. see the type from xverse-api repo

Comment (teebszet) [reply api/xverse.ts:114]: need to extend BaseToken as created from jordan's PR https://github.com/secretkeylabs/xverse-core/pull/353

Comment (abdulhaseeb4239) [reply api/xverse.ts]: the BaseToken has `image` that we don't have in this API's response. I've added the response type

File: api/xverse.ts [hunk 2]
```diff
@@ -92,6 +93,34 @@ export async function getCoinsInfo(
...
+export async function getBrc20Tokens(
+  network: NetworkType,
+  tickers: string[],
+  fiatCurrency: string,
+): Promise<Brc20TokensResponse | null> {
+  const url = `${XVERSE_API_BASE_URL(network)}/v1/brc20/tokens`;
+
+  const params = {
+    currency: fiatCurrency,
+    tickers: tickers,
```
Comment (teebszet) [inline api/xverse.ts:111]: note: need to double check that this gets serialised into the query params properly

File: types/api/xverse/coins.ts [hunk 3]
```diff
@@ -15,3 +15,12 @@ export interface SignedUrlResponse {
 }
 
 export type CoinsResponse = Coin[];
+
+export type Brc20Token = {
```
Comment (teebszet) [inline types/api/xverse/coins.ts:19]: did you consider extending BaseToken like Coin does?
```suggestion
export interface Brc20Token extends BaseToken = {
```

File: api/xverse.ts (hunk 2 above)
Comment (teebszet) [reply api/xverse.ts:111]: oh yeah this format is not supported:

http://localhost:3001/v1/brc20/tokens?currency=USD&tickers[]=TTIN&tickers[]=SA

File: types/api/xverse/coins.ts (hunk 3 above)
Comment (jordankzf) [reply types/api/xverse/coins.ts]: `ticker` is optional in `BaseToken` but non-optional in `Brc20Token`. Might have to redeclare it.

**Review by teebszet — APPROVED** [review]

File: api/xverse.ts (hunk 1 above)
Comment (teebszet) [reply api/xverse.ts]: think it's OK since image is optional on BaseToken

File: types/api/xverse/coins.ts (hunk 3 above)
Comment (teebszet) [reply types/api/xverse/coins.ts]: yup done

**Pull request merged**